metaserver/src/main/scala/scala/meta/languageserver/ScalametaLanguageServer.scala

```diff
@@ -449,6 +442,18 @@ object ScalametaLanguageServer extends LazyLogging {
     subscriber -> semanticdbPublisher
   }
 
+  def configurationStream(connection: Connection)(
+    implicit scheduler: Scheduler
+  ): (Observer.Sync[JsValue], Observable[Configuration]) = {
+    val (subscriber, publisher) = multicast[JsValue]
+    val configurationPublisher = publisher
+      .map(json => (json \ "scalameta").as[Configuration])
+      .doOnNext(conf => logger.info(s"Configuration updated $conf"))
+      .doOnError(e => connection.showMessage(MessageType.Error, e.getMessage))
```

Comment:
I think Observable stops on the first error so this is not enough to continue streaming configuration. I tried
```scala
    import monix.execution.Scheduler.Implicits.global
    val o = Observable
      .fromIterable(1.to(10))
      .map {
        case 5 => sys.error("5")
        case n => n + 1
      }
      .doOnError(e => println(s"EREEEEEEEEEE $e"))
    o.foreach(println)
```
and it prints out

```
2
3
4
5
EREEEEEEEEEE java.lang.RuntimeException: 5
```

I think it's best to avoid exceptions and do something like this instead
```scala
    import monix.execution.Scheduler.Implicits.global
    val o = Observable
      .fromIterable(1.to(10))
      .map {
        case 5 => Left(5)
        case n => Right(n)
      }
      .doOnNext {
        case Left(err) => println(s"FAIL $err")
        case _ => ()
      }
      .collect {
        case Right(x) => x
      }
    o.foreach(println)
```

which prints
```
1
2
3
4
FAIL 5
6
7
8
9
10
```

Alternatively, 

```scala
    val o = Observable
      .fromIterable(1.to(10))
      .liftByOperator[Int] { out =>
        new Subscriber[Int] {
          override implicit def scheduler: Scheduler = out.scheduler
          override def onError(ex: Throwable): Unit = out.onError(ex)
          override def onComplete(): Unit = out.onComplete()
          override def onNext(elem: Int): Future[Ack] = elem match {
            case 5 =>
              println("FAIL 5")
              Ack.Continue
            case n =>
              out.onNext(n)
          }
        }
      }
    o.foreach(println)
```
which prints the same.

Reply:
very nice catch, thank you!